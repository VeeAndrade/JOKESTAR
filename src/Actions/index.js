export const updateUser = (username) => ({
  type: 'UPDATE_USER',
  username
});

export const setJoke = (joke) => ({
  type: 'SET_JOKE',
  joke
});

export const favoriteJoke = (joke) => ({
  type: 'FAVORITE_JOKE',
  joke
});

export const addFetchParams = (stateObj) => ({
  type: 'ADD_PARAMS',
  stateObj
});