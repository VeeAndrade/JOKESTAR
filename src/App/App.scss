.main-app {
  background: linear-gradient(rgba(49, 48, 48, 0.603), rgba(54, 52, 52, 0.527)),url('../Assets/moritz-mentges-2RLRbRkWtpc-unsplash.jpg');
  width: 100%;
  height: 60rem;
  background-size: contain;
  font-family: 'Questrial';
  font-weight: 400;
  display: flex;
  flex-direction: column;
  align-items: center;
}