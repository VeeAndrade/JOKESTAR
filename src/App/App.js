import React, { Component } from 'react';
import Nav from '../Nav/Nav';
import Login from '../Login/Login';
import MainPage from '../MainPage/MainPage';
import LandingPage from '../LandingPage/LandingPage';
import { Route } from 'react-router-dom';
import { updateUser } from '../Actions'
import { connect } from 'react-redux';
import './App.scss';

export class App extends Component {
  constructor() {
    super();
  }

  logoutUser = () => {
    this.props.updateUser('')
  }

  render() {
    return (
      <main className='main-app'>
        <Nav logoutUser={this.logoutUser}/>
        <Route exact path='/'>
          <LandingPage />
        </Route>
        <Route exact path='/main-page' render={() => <MainPage />}/>
        <Route exact path='/login' render={() => <Login />}/>
      </main>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  updateUser: username => dispatch(updateUser(username))
})

export default connect(null, mapDispatchToProps)(App);