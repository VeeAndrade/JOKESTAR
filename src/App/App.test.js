import React from 'react';
import { App, mapDispatchToProps } from './App';
import { updateUser } from '../Actions';
import { shallow } from 'enzyme';


describe('App', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<App />)
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapDispatchToProps', () => {
  it('should call updateUser when logout user is called', () => {
    let mockResponse = {username: 'EvilKitty'}
    let mockDispatch = jest.fn();
    let actionToDispatch = updateUser(mockResponse)
    let mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.updateUser(mockResponse)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  });
});
