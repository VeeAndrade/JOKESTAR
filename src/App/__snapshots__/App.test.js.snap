// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<main
  className="main-app"
>
  <Connect(Nav)
    logoutUser={[Function]}
  />
  <Route
    exact={true}
    path="/"
  >
    <Connect(LandingPage) />
  </Route>
  <Route
    exact={true}
    path="/main-page"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/login"
    render={[Function]}
  />
</main>
`;
